### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by LDVera**
some files was modified to get the login functionality
in the assets folder was addes an icon and the background of the page (pending to change)
some migrations was added to add tables and some columns in existing tables
the root now shows the login page and any user that wanna go enter the admin page needs
get the admin role with rolify
in this version the functionality of generate tokens was added too.

# Description
What did you implemented/Why did you implemented this:
 - the admin users needs get and space to control the information 
 - if the app have many admin users we need to control how can join to the admin panel
 - as second the api functionality was added too, to generate tokens and use with the mobile app

### CODE REVIEW

File: app/controllers/api/v1/application_api_controller.rb [hunk 1]
```diff
@@ -1,6 +1,55 @@
...
+  def current_user
+    @current_user ||= super || User.find(@current_user_id)
+  end
+
+  # check that authenticate_user has successfully returned @current_user_id (user is authenticated)
+  def signed_in?
+    @current_user_id.present?
+  end
+
+  def token_is_available?(jwt_payload)
```
Comment (MiguelOcegueraM) [inline app/controllers/api/v1/application_api_controller.rb:51]: nice method

File: app/controllers/api/v1/users/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class  API::V1::Users::SessionsController < Devise::SessionsController
+  skip_before_action :verify_authenticity_token
+  
+  respond_to :json
+  # before_action :configure_sign_in_params, only: [:create]
+  # GET /resource/sign_in
```
Comment (MiguelOcegueraM) [inline app/controllers/api/v1/users/sessions_controller.rb:8]: remember, delete all the commented code that you aren't using

File: config/initializers/devise.rb [hunk 3]
```diff
@@ -15,7 +15,17 @@
   # Devise will use the `secret_key_base` as its `secret_key`
   # by default. You can change it below and use your own secret key.
   config.secret_key = '2e20dcc6110f16d8b4a26cae4389d33be10464b44df7bb13f0769d8d61533c3489cf7082af2129eb9505f6c85912aa0ea9abaa560399c15c07102a63ec32d2cf'
-
+  config.jwt do |jwt|
+    jwt.secret = '2e20dcc6110f16d8b4a26cae4389d33be10464b44df7bb13f0769d8d61533c3489cf7082af2129eb9505f6c85912aa0ea9abaa560399c15c07102a63ec32d2cf'
```
Comment (ArielMD) [inline config/initializers/devise.rb:19]: You can try using environment variables so you don't show credentials

**Review by ArielMD — APPROVED** [review]
good job, that's fine with me

File: app/views/devise/sessions/_new.html.erb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                Remember Me
+              </label>
+            </div>
+          </div>
+          <!-- /.col -->
+          <div class="col-4">
+            <%= f.submit "Log in", class: "btn btn-primary btn-block" %>
+          </div>
+          <!-- /.col -->
+        </div>
```
Comment (MiguelOcegueraM) [inline app/views/devise/sessions/_new.html.erb:43]: In my opinion there is a lot of divs, try to use more tags in your html for better structure.

File: app/views/layouts/_main.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        </div><!-- /.row -->
+      </div><!-- /.container-fluid -->
+    </div>
+
+    <div class="content container-fluid">
+      <div class="container-fluid">
+        <%= yield %>
+      </div>
+    </div>
+  </div>
```
Comment (MiguelOcegueraM) [inline app/views/layouts/_main.html.erb:27]: same here, a lot of divs

**Review by MiguelOcegueraM — APPROVED** [review]
Nice code mates but I think you need to improve your html.
Read about the semantic HTML in this blog it can help you
https://dev.to/kenbellows/stop-using-so-many-divs-an-intro-to-semantic-html-3i9i

**Review by LPMendoza — APPROVED** [review]
Good job man, I like the code in your backend.

**Pull request merged**